sw/host/opentitanlib/src/transport/hyperdebug/spi.rs

```diff
@@ -525,6 +759,83 @@ impl Target for HyperdebugSpiTarget {
+            return eeprom::default_run_eeprom_transactions(self, transactions);
+        }
+        let mut stream_state = StreamState::NoPending;
+        loop {
+            match transactions {
+                [eeprom::Transaction::Command(pre_cmd), eeprom::Transaction::Write(cmd, ref wbuf), eeprom::Transaction::WaitForBusyClear, rest @ ..] =>
+                {
+                    let write_enable = if pre_cmd.get_opcode() == [0x06] {
+                        // Write enable is done by eeprom_write_transaction()
+                        true
+                    } else {
+                        self.run_transaction(&mut [Transfer::Write(cmd.to_bytes()?)])?;
+                        false
+                    };
+                    stream_state = self.eeprom_transmit(
+                        write_enable,
+                        cmd,
+                        wbuf,
+                        &mut [],
+                        true, /* wait_for_busy_clear */
+                        stream_state,
+                    )?;
+                    transactions = rest;
+                }
+                [eeprom::Transaction::Command(cmd), rest @ ..] => {
+                    stream_state =
+                        self.eeprom_transmit(false, cmd, &[], &mut [], false, stream_state)?;
+                    transactions = rest;
+                }
+                [eeprom::Transaction::Read(cmd, ref mut rbuf), rest @ ..] => {
+                    stream_state =
+                        self.eeprom_transmit(false, cmd, &[], rbuf, false, stream_state)?;
+                    transactions = rest;
+                }
+                [eeprom::Transaction::Write(cmd, ref wbuf), eeprom::Transaction::WaitForBusyClear, rest @ ..] =>
+                {
+                    stream_state = self.eeprom_transmit(
+                        false, /* write_enable */
+                        cmd,
+                        wbuf,
+                        &mut [],
+                        true, /* wait_for_busy_clear */
+                        stream_state,
+                    )?;
+                    transactions = rest;
+                }
+                [eeprom::Transaction::Write(cmd, ref wbuf), rest @ ..] => {
+                    stream_state =
+                        self.eeprom_transmit(false, cmd, wbuf, &mut [], false, stream_state)?;
+                    transactions = rest;
+                }
+                [eeprom::Transaction::WaitForBusyClear, rest @ ..] => {
+                    self.get_last_streamed_data(stream_state)?;
+                    while {
+                        let mut buf = [0u8; 1];
+                        self.run_transaction(&mut [
+                            Transfer::Write(&[eeprom::READ_STATUS]),
+                            Transfer::Read(&mut buf),
+                        ])?;
+                        buf[0]
+                    } & eeprom::STATUS_WIP
+                        != 0
+                    {}
```

Comment:
This while loop can be cleaned up similarly to the prior comment.

Reply:
Done